doc/source/conf.py

```diff
@@ -54,6 +54,23 @@
 seqdiag_fontpath = os.path.abspath('./source/fonts/DejaVuSerif.ttf')
 
 
+def _skip_pyqtsignal_doc(app, what, name, obj, skip, options):
+    # see https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#event-autodoc-skip-member
+    if what == "class" and "pyqtSignal" in obj.__class__.__name__:
+        return True
+    # fallback to other handlers or autodoc default
+    return None
+
+
+def setup(app):
+    """Global setup function for sphinx documentation.
+
+    Connect to autodoc event so invalid pyqtSignal docs can be skipped.
```

Comment:
We ran across this in our project as well, and found your workaround here.

But I'm curious, what is it in `pyqtSignal`'s docstring that is invalid? I'm staring at it, comparing it to e.g. that for `pyqtSlot`, but I can't see what's special about it :p

Reply:
Oh, it's been a while and I don't quite remember exactly. I think on certain versions of Python and certain versions of PyQt, the pyqtSignal's `__doc__` property was set to something weird or caused a segfault or something when it was read. We, as users of pyqtsignal, can't set the docstring through normal means as far as I know so this was necessary. We could make a PR and try removing this skip logic and see what happens. Let me try it locally first...

Reply:
Ah, but I confirmed that your workaround works for us. We ran into the problem on Python 3.10, using 5.15.2.

But I'm print the `__doc__` of `pyqtSignal`, and I can't see the reason why sphinx is complaining:

```
(env) orexplore@vidar-m7:~$ python3 -c 'from PyQt5.QtCore import pyqtSignal; print(pyqtSignal.__doc__)'
pyqtSignal(*types, name: str = ..., revision: int = ..., arguments: Sequence = ...) -> PYQT_SIGNAL

types is normally a sequence of individual types.  Each type is either a
type object or a string that is the name of a C++ type.  Alternatively
each type could itself be a sequence of types each describing a different
overloaded signal.
name is the optional C++ name of the signal.  If it is not specified then
the name of the class attribute that is bound to the signal is used.
revision is the optional revision of the signal that is exported to QML.
If it is not specified then 0 is used.
arguments is the optional sequence of the names of the signal's arguments.

(env) orexplore@vidar-m7:~$
```

Reply:
Ok, got it. So this was added because when sphinx docs are generated it produces a ton of these warnings for each pyqtsignal:

```
docstring of uwsift.queue.TaskQueue.didMakeProgress:13: WARNING: Inline emphasis start-string without end-string.
```

This is what the docstring for all pyqtsignals looks like:

```python
In [6]: Document.didAddBasicLayer.__doc__
Out[6]: "pyqtSignal(*types, name: str = ..., revision: int = ..., arguments: Sequence = ...) -> PYQT_SIGNAL\n\ntypes is normally a sequence of individual types.  Each type is either a\ntype object or a string that is the name of a C++ type.  Alternatively\neach type could itself be a sequence of types each describing a different\noverloaded signal.\nname is the optional C++ name of the signal.  If it is not specified then\nthe name of the class attribute that is bound to the signal is used.\nrevision is the optional revision of the signal that is exported to QML.\nIf it is not specified then 0 is used.\narguments is the optional sequence of the names of the signal's arguments.\n"
```

I don't like generating sphinx docs that contain warnings so I typically add `-W` to make it fail when it encounters warnings.

I just saw your new comment...one sec...

Reply:
It is the `*` in `*types` that in RST starts a bolded text, but it is never enclosed.

I chose this simple workaround rather than forcing myself to modify ever pyqtsignal object and all future signals. I was hoping for something easier or better, but couldn't come up with it at the time.

Reply:
Alright, yea I was just wondering why we're seeing this for pyqtSignal and not pyqtSlot, which we're also using and which also has an asterisk there.

Reply:
(same here we treat warnings as errors and that's how we discovered the issue. It only appears on Python 3.10, not on 3.8 which we also test)

Reply:
I'll doublecheck what the docstring looks like for our pyqtSlot-decorated functions, perhaps the pyqtSlot docstring doesn't propagate the same way the pyqtSignal one does.

Reply:
That or wrap it in double backticks to represent a literal.